pennylane/measurements/counts.py

```diff
@@ -153,7 +161,13 @@ def circuit():
             )
         wires = Wires(wires)
 
-    return CountsMP(obs=op, wires=wires, all_outcomes=all_outcomes)
+    if eigvals is not None and len(wires) > 1:
+        raise ValueError(
+            "If eigvals are provided instead of an observable, specifying more than one wire "
+            "is not supported."
```

Comment:
This raises an error if wires is an integer, which is an allowed type according to the docstring (and previous behaviour), so I think we should convert to a `Wires` object so we can safely take the length regardless of input type. 

Wrt the error message, saying "instead of an observable" sounds like it means more than one wire *is* supported with an observable, which isn't true and is a bit confusing. 

Finally, I guess we may as well also catch the case where there are `eigvals` but no wires, since that's also not allowed and gives a hard-to-parse error message currently. We can just catch too many and too few in one go with something like `if eigvals and wires is None or len(Wires(wires))>1:`. The error message could just be something like `"When sampling with eigvals, a single wire must be specified"` so that it makes sense with both too many and too few wires.

Reply:
I will update the error message. Although `wires` would have already been converted to `Wires(wires)` above by the time the code reaches here so specifying `wires` with an integer would not raise an error.